.github/workflows/license.yml

```diff
@@ -0,0 +1,51 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: 'License Check'
+on: [pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@v4
+      - name: 'Check license header'
+        uses: apache/skywalking-eyes@main
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@v4
+        with:
+          vulnerability-check: false
+          license-check: true
+          # Incompatible licenses addressed here: https://www.apache.org/legal/resolved.html
+          # Special notice for GPL licenses: https://www.apache.org/licenses/GPL-compatibility.html
+          # Find SPDX identifiers here: https://spdx.org/licenses/
+          deny-licenses: >
+            MS-LPL, BUSL-1.1,
+            CC-BY-NC-1.0, CC-BY-NC-2.0, CC-BY-NC-2.5, CC-BY-NC-3.0, CC-BY-NC-4.0,
+            GPL-1.0, GPL-2.0, GPL-3.0, AGPL-3.0, LGPL-2.0, LGPL-2.1, LGPL-3.0,
+            GPL-1.0-only, GPL-2.0-only, GPL-3.0-only, AGPL-3.0-only, LGPL-2.0-only, LGPL-2.1-only, LGPL-3.0-only,
+            QPL-1.0, Sleepycat, SSPL-1.0, CPOL-1.02,
+            BSD-4-Clause, BSD-4-Clause-UC, NPL-1.0, NPL-1.1, JSON
```

Comment:
Your dependencies use 15-20 licenses counting `OR` expressions.

I think that a whitelist that only contains the licenses that you **actually** use would be shorter and safer:

* SPDX list has more than 600 known licenses and only a small part of them has been classified by the ASF,
* you can always update this workflow if a new kind of license appears among your dependencies.

Reply:
>You can always update this workflow if a new kind of license appears among your dependencies.

The purpose of my proposed PR is to remove `known-dependencies.txt`. For contributors, adding a license in the GitHub Actions workflow file after encountering CI failures and reviewing logs is significantly more challenging than adding an artifact in `known-dependencies.txt`. It also increases the level of review required for PRs (which needs PMC decision, but unfortunately, we only have one PMC available to review PRs).

The SPDX list is not exhaustive, and we would need to invest a considerable amount of time in adding exceptions in the GitHub Actions workflow file. Utilizing a license whitelist increases the manual workload for developers, which is not something we want to spend time on. Instead, utilizing a license blacklist can prevent the introduction of incompatible licenses in everyday community contributions without requiring any configuration from developers. This is also what the `apache/skywalking-eyes/dependency` action has been doing all along, but unfortunately, it does not support Gradle.

During the release process, the Release Manager will need to double-check the outputs of the `generateDistLicense` task and review all licenses for compatibility once again. I believe this would be entirely sufficient.

Reply:
> The purpose of my proposed PR is to remove `known-dependencies.txt`. For contributors, adding a license in the GitHub Actions workflow file after encountering CI failures and reviewing logs is significantly more challenging than adding an artifact in `known-dependencies.txt`. It also increases the level of review required for PRs (which needs PMC decision, but unfortunately, we only have one PMC available to review PRs).

The process of adding a new dependency by an external contributor doesn't have to be easy.

Submitting a PR with a new dependency is a one time effort, while you will have to maintain that code and upgrade the dependency for years.

Reply:
>The process of adding a new dependency by an external contributor doesn't have to be easy.

Yes, I agree with your perspective. Whether it's maintaining a dependency whitelist or a license whitelist, neither task is easy. Maintaining a dependency blacklist is impractical, but maintaining a license blacklist essentially requires a one-time effort. That's why I believe a license blacklist would be more suitable for our community~ ☺️